sveltekit-directus/index.md

```diff
@@ -0,0 +1,348 @@
+	directus.deleteItem = async (collection, id) => directus.request(deleteItem(collection, id));
+
+	return directus;
+}
+
+export default getDirectusInstance;
+```
+
+In order to make this work we also need to create a `hooks.server.js` file with the following content in the root
+directory. It makes sure that the required headers for fetching JavaScript content are returned by the SvelteKit Server
+
+```js
+export async function handle({ event, resolve }) {
+	return await resolve(event, {
+		filterSerializedResponseHeaders: (key, value) => {
+			return key.toLowerCase() === 'content-type';
+		},
+	});
+}
+```
+
+::: tip Directus SDK
+
+Theoretically you could also make HTTP requests to your Directus server endpoint directly via SvelteKit's `fetch`
+implementation. However the Directus SDK offers some nice [additional features](/guides/sdk/getting-started).
+
+:::
+
+Also create the environment variable inside a `.env` file in the root directory
+
+```js
+VITE_APIURL = 'https://directus.example.com';
+```
+
+Ensure your API URL is correct when initializing the Directus JavaScript SDK.
+
+## Setup Global Metadata and Settings
+
+In your Directus project, navigate to Settings -> Data Model and create a new collection called `global`. Under the
+Singleton option, select 'Treat as a single object', as this collection will have just a single entry containing global
+website metadata.
+
+Create two text input fields - one with the key of `title` and one `description`.
+
+Navigate to the content module and enter the global collection. Collections will generally display a list of items, but
+as a singleton, it will launch directly into the one-item form. Enter information in the title and description field and
+hit save.
+
+![A form named "Global" has two inputs - a title and a description, each filled with some text.](global-config.webp)
+
+By default, new collections are not accessible to the public. Navigate to Settings -> Roles & Permissions -> Public and
+give Read access to the Global collection.
+
+## Prepare SvelteKit to use Directus instance
+
+Create a new file called `+page.js` in the root directory along the `.page.svelte` file. This file's load function will
+be responsible to fetch the data on the client and on the server during Server Side Rendering.
+
+```js
+/** @type {import('./$types').PageLoad} */
+import getDirectusInstance from '$lib/directus';
+export async function load({ fetch }) {
+	const directus = getDirectusInstance(fetch);
+	return {
+		global: await directus.readItems('global'),
+	};
+}
+```
+
+Correspondingly we need to modify the `+page.svelte` file to take the data into account and display it on our site.
+SvelteKit magically populates the data property and also updates it should any dependency(variable, url, etc.) change
+which influences the data fetch mechanism in the `+page.js` file. Currently there is no dependency, later you will see
+how this works.
```

Comment:
```suggestion
Modify the `+page.svelte` file to use the new data and display it on our site:
```

Remove half explaining things in advance. It doesn't make sense yet, so save it for when it does.

Reply:
moved it to a later section and shortened it. would include it however, as it is quite important to understand SSR